src/main/capi/duckdb_value-c.cpp

```diff
@@ -39,3 +52,54 @@ int64_t duckdb_get_int64(duckdb_value value) {
 	}
 	return duckdb::BigIntValue::Get(*val);
 }
+
+duckdb_value duckdb_create_struct_value(duckdb_logical_type type, duckdb_value *values) {
+	if (!type || !values) {
+		return nullptr;
+	}
+	const auto &ltype = UnwrapType(type);
+	if (ltype.id() != duckdb::LogicalTypeId::STRUCT) {
+		return nullptr;
+	}
+	auto count = duckdb::StructType::GetChildCount(ltype);
+
+	duckdb::vector<duckdb::Value> unwrapped_values;
+	for (idx_t i = 0; i < count; i++) {
+		auto value = values[i];
+		if (!value) {
+			return nullptr;
+		}
+		unwrapped_values.emplace_back(UnwrapValue(value));
+	}
+	duckdb::Value *struct_value = new duckdb::Value;
+	try {
+		*struct_value = duckdb::Value::STRUCT(ltype, unwrapped_values);
+	} catch (...) {
+		delete struct_value;
+		return nullptr;
+	}
+	return WrapValue(struct_value);
```

Comment:
When this takes a unique_ptr, we also don't need to explicitly use `new` and `delete` here